checks/default.nix

```diff
@@ -107,6 +107,23 @@ in
     };
   });
 
+  chainedMultiple = pkgs.linkFarmFromDrvs "chainedMultiple" (map
```

Comment:
All these checks are not going to catch any issues. The only reason CI in Fedimint catches the problem is because we have a `cargo/rustc` wrapper that forbids rebuilds at certain point of the chain. Otherwise bugs w.r.t. target inheritance etc. just cause slow rebuilds which is easy to overlook.

Reply:
You might just want to copy&paste this wrapper into your testing suite ... . It comes down to https://github.com/fedimint/fedimint/blob/97b3b701aed2dffc556edd12c2159afcf9aa526d/flake.toolchain.nix#L130 and https://github.com/fedimint/fedimint/tree/master/scripts/dev/rustc-wrapper .

It seems that testing that rustc doesn't rebuild stuff is quite important and common.

Reply:
Ah thanks for the suggestion! We've got a couple "do-not-rebuild" tests but they're pretty manual and annoying to deal with.

Will take a look at incorporating this

Reply:
@dpc have some questions about how your wrapper script works: i see there's an environment variable which can be flipped to deny any new compilation, makes sense, but how to you check that there are no rebuilds between the `deps-only` and the actual package build?

Example:
1. build deps only with the wrapper, env var not set things build from scratch for the first time
2. build the real workspace source with artifacts from 1, env var not set since we still need to build the workspace
3. Set the env var, rebuild the workspace package with artifacts from 2, nothing should build

This example makes sense to me, but I'm curious if you do (or if you could) check that there aren't any extra rebuilds between steps 1 and 2

Reply:
> but how to you check that there are no rebuilds between the `deps-only` and the actual package build?

I don't use it like that. I only use it after I have built whole workspace and I run derivations for tests.

Reply:
Updated, thanks again for the suggestion!